pkg/controllers/gateway/gateway_controller.go

```diff
@@ -133,6 +134,10 @@ func (r *GatewayReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ct
 	}
 
 	if !controllerutil.ContainsFinalizer(upstreamGateway, GatewayFinalizer) {
+		// The first time we see the gateway we want to make sure a DNSPolicy exists for it
```

Comment:
wouldn't be easier and more robust to check for DNSPolicy back references in the gateway and when there is none, then create a default DNS policy?

Reply:
The gateway can be targeted by a single policy. If you implement based on back references as I suggest, deleting the default policy will trigger a new policy to be created. Users would have to update the default policy instead of deleting existing default one and create a new one. 

If you want otherwise, then LGTM.

Reply:
Yes, so when i discussed this with @maleck13  we didn't, for reasons i can't remember, want them to have to update the default one we were creating. It really depends what we actually want to happen here, have us make sure a policy always exist and users just update that one, or create one initially and allow users to delete that and create their own.

I'll leave it as is for now, we can always change it later.

Reply:
The reason for not forcing the use of the existing policy was to allow users to create a new policy resource and store it in git and use something like argo to sync state. 
The user flows here are:
1) Update the existing policy 
2) create a new policy and then delete the existing policy